packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayer.m

```diff
@@ -245,56 +236,41 @@ - (void)observeValueForKeyPath:(NSString *)path
                       ofObject:(id)object
                         change:(NSDictionary *)change
                        context:(void *)context {
+  NSAssert([NSThread isMainThread], @"event sink must only be accessed from the main thread");
+  if (!_eventSink) {
+    return;
+  }
   if (context == timeRangeContext) {
-    if (_eventSink != nil) {
-      NSMutableArray<NSArray<NSNumber *> *> *values = [[NSMutableArray alloc] init];
-      for (NSValue *rangeValue in [object loadedTimeRanges]) {
-        CMTimeRange range = [rangeValue CMTimeRangeValue];
-        int64_t start = FVPCMTimeToMillis(range.start);
-        [values addObject:@[ @(start), @(start + FVPCMTimeToMillis(range.duration)) ]];
-      }
-      _eventSink(@{@"event" : @"bufferingUpdate", @"values" : values});
+    NSMutableArray<NSArray<NSNumber *> *> *values = [[NSMutableArray alloc] init];
+    for (NSValue *rangeValue in [object loadedTimeRanges]) {
+      CMTimeRange range = [rangeValue CMTimeRangeValue];
+      int64_t start = FVPCMTimeToMillis(range.start);
+      [values addObject:@[ @(start), @(start + FVPCMTimeToMillis(range.duration)) ]];
     }
+    _eventSink(@{@"event" : @"bufferingUpdate", @"values" : values});
   } else if (context == statusContext) {
     AVPlayerItem *item = (AVPlayerItem *)object;
     switch (item.status) {
-      case AVPlayerItemStatusFailed:
-        [self sendFailedToLoadVideoEvent];
-        break;
       case AVPlayerItemStatusUnknown:
         break;
       case AVPlayerItemStatusReadyToPlay:
         [item addOutput:_videoOutput];
```

Comment:
If status is changed to `AVPlayerItemStatusReadyToPlay` before the event sink is set up then it seems this `addOutput` will not be called.

Reply:
Oops thanks for catching this!